app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderInfo.kt

```diff
@@ -0,0 +1,401 @@
+package com.android.solvit.seeker.ui.provider
+
+import android.widget.Toast
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material.icons.filled.Share
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.AsyncImage
+import com.android.solvit.R
+import com.android.solvit.seeker.model.provider.ListProviderViewModel
+import com.android.solvit.shared.model.provider.Provider
+import com.android.solvit.shared.model.review.Review
+import com.android.solvit.shared.model.review.ReviewViewModel
+import com.android.solvit.shared.ui.navigation.NavigationActions
+
+@Composable
+fun ProviderInfoScreen(
+    navigationActions: NavigationActions,
+    providerViewModel: ListProviderViewModel = viewModel(factory = ListProviderViewModel.Factory),
+    reviewsViewModel: ReviewViewModel = viewModel(factory = ReviewViewModel.Factory)
+) {
+  val provider = providerViewModel.selectedProvider.collectAsState().value ?: return
```

Comment:
Error handling here is not very clear. What would happen if the Provider is null? The behavior should probably be an error message and going back to Select Provider screen

Reply:
You would just get a white screen but each navigation to the ProviderInfoScreen from other screens is preceded by a call to listProviderViewModel.selectProvider(provider) which ensures the fact that the selectedProvider will not be null when someone has to navigate to the ProviderInfoScreen.